File: Samples/Basic/DocumentReader-sample/app/src/main/java/com/regula/documentreader/SettingsFragment.java
package com.regula.documentreader;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import android.widget.RadioButton;
import android.widget.RadioGroup;

import androidx.fragment.app.Fragment;

public class SettingsFragment extends Fragment {
    private RadioButton customRfidRb;
    private RadioButton defaultRfidRb;

    private RadioButton customCameraRb;
    private RadioButton customCamera2Rb;
    private RadioButton customCameraRegRb;
    private RadioButton defaultCameraRb;

    private volatile MainFragment.MainCallbacks mCallbacks;
    public static boolean isCustomRfidSelected;
    public static int RFID_RESULT = 100;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.fragment_settings, container, false);
        customRfidRb = root.findViewById(R.id.customRfidReaderRb);
        defaultRfidRb = root.findViewById(R.id.defaultRfidReaderRb);

        defaultCameraRb = root.findViewById(R.id.defaultScannerRb);
        customCameraRb = root.findViewById(R.id.customCameraRb);
        customCamera2Rb = root.findViewById(R.id.customCamera2Rb);
        customCameraRegRb = root.findViewById(R.id.customRegCameraRb);

        checkCameraByMode(((BaseActivity) getActivity()).activeCameraMode);
        checkRfidByMode(((BaseActivity) getActivity()).rfidMode);
        setupRfid();
        setupCameraRbListeners();
        return root;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        mCallbacks = (MainFragment.MainCallbacks) getActivity();
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mCallbacks = null;
    }

    public void setupRfid() {
        defaultRfidRb.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked)
                ((BaseActivity) getActivity()).rfidMode = RfidMode.DEFAULT;
        });
        customRfidRb.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked)
                ((BaseActivity) getActivity()).rfidMode = RfidMode.CUSTOM;
        });
    }

    public void setupCameraRbListeners() {
        defaultCameraRb.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked)
                ((BaseActivity) getActivity()).activeCameraMode = CameraMode.SDK_SHOW_SCANNER;
        });
        customCameraRb.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked)
                ((BaseActivity) getActivity()).activeCameraMode = CameraMode.SHOW_CUSTOM_CAMERA_ACTIVITY;
        });
        customCamera2Rb.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked)
                ((BaseActivity) getActivity()).activeCameraMode = CameraMode.SHOW_CUSTOM_CAMERA_ACTIVITY2;
        });
        customCameraRegRb.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked)
                ((BaseActivity) getActivity()).activeCameraMode = CameraMode.SHOW_CAMERA_ACTIVITY;
        });
    }

    private void checkCameraByMode(int camMode) {
        switch (camMode) {
            case CameraMode.SDK_SHOW_SCANNER:
                defaultCameraRb.setChecked(true);
                break;
            case CameraMode.SHOW_CAMERA_ACTIVITY:
                customCameraRegRb.setChecked(true);
                break;
            case CameraMode.SHOW_CUSTOM_CAMERA_ACTIVITY:
                customCameraRb.setChecked(true);
                break;
            case CameraMode.SHOW_CUSTOM_CAMERA_ACTIVITY2:
                customCamera2Rb.setChecked(true);
                break;
        }
    }

    private void checkRfidByMode(int rfidMode) {
        switch (rfidMode) {
            case RfidMode.DEFAULT:
                defaultRfidRb.setChecked(true);
                break;
            case RfidMode.CUSTOM:
                customRfidRb.setChecked(true);
                break;
        }
    }

    public static class RfidMode {
        public static final int DEFAULT = 0;
        public static final int CUSTOM = 1;
    }

    private static class CameraMode {
        public static final int SDK_SHOW_SCANNER = 0;
        public static final int SHOW_CAMERA_ACTIVITY = 1;
        public static final int SHOW_CUSTOM_CAMERA_ACTIVITY = 2;
        public static final int SHOW_CUSTOM_CAMERA_ACTIVITY2 = 3;
    }
}

File: Samples/Basic/DocumentReader-sample_kotlin/app/src/main/java/com/regula/documentreader/custom/SettingsFragment.kt
package com.regula.documentreader.custom

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CompoundButton
import android.widget.RadioButton
import android.widget.RadioGroup
import androidx.fragment.app.Fragment
import com.regula.documentreader.BaseActivity
import com.regula.documentreader.MainFragment.MainCallbacks
import com.regula.documentreader.R

class SettingsFragment : Fragment() {
    private var cameraRg: RadioGroup? = null
    private var customRfidRb: RadioButton? = null
    private var defaultRfidRb: RadioButton? = null

    private var customCameraRb: RadioButton? = null
    private var customCamera2Rb: RadioButton? = null
    private var customCameraRegRb: RadioButton? = null
    private var defaultCameraRb: RadioButton? = null

    private var mCallbacks: MainCallbacks? = null
    var isCustomRfidSelected = false
    var RFID_RESULT = 100

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val root: View = inflater.inflate(R.layout.fragment_settings, container, false)
        cameraRg = root.findViewById(R.id.customCameraRg)
        customRfidRb = root.findViewById(R.id.customRfidReaderRb)
        defaultRfidRb = root.findViewById(R.id.defaultRfidReaderRb)
        defaultCameraRb = root.findViewById(R.id.defaultScannerRb)
        customCameraRb = root.findViewById(R.id.customCameraRb)
        customCamera2Rb = root.findViewById(R.id.customCamera2Rb)
        customCameraRegRb = root.findViewById(R.id.customRegCameraRb)
        (activity as BaseActivity?)?.let { checkCameraByMode(it.activeCameraMode) }
        (activity as BaseActivity?)?.let { checkRfidByMode(it.rfidMode) }
        setupRfid()
        setupCameraRbListeners()
        return root
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mCallbacks = activity as MainCallbacks?
    }

    override fun onDetach() {
        super.onDetach()
        mCallbacks = null
    }

    fun setupRfid() {
        defaultRfidRb!!.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            if (isChecked) (activity as BaseActivity?)!!.rfidMode =
                RfidMode.DEFAULT
        }
        customRfidRb!!.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            if (isChecked) (activity as BaseActivity?)!!.rfidMode =
                RfidMode.CUSTOM
        }
    }

    fun setupCameraRbListeners() {
        defaultCameraRb!!.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            if (isChecked) (activity as BaseActivity?)!!.activeCameraMode =
                CameraMode.SDK_SHOW_SCANNER
        }
        customCameraRb!!.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            if (isChecked) (activity as BaseActivity?)!!.activeCameraMode =
                CameraMode.SHOW_CUSTOM_CAMERA_ACTIVITY
        }
        customCamera2Rb!!.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            if (isChecked) (activity as BaseActivity?)!!.activeCameraMode =
                CameraMode.SHOW_CUSTOM_CAMERA_ACTIVITY2
        }
        customCameraRegRb!!.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            if (isChecked) (activity as BaseActivity?)!!.activeCameraMode =
                CameraMode.SHOW_CAMERA_ACTIVITY
        }
    }

    private fun checkCameraByMode(camMode: Int) {
        when (camMode) {
            CameraMode.SDK_SHOW_SCANNER -> defaultCameraRb!!.isChecked = true
            CameraMode.SHOW_CAMERA_ACTIVITY -> customCameraRegRb!!.isChecked = true
            CameraMode.SHOW_CUSTOM_CAMERA_ACTIVITY -> customCameraRb!!.isChecked = true
            CameraMode.SHOW_CUSTOM_CAMERA_ACTIVITY2 -> customCamera2Rb!!.isChecked = true
        }
    }

    private fun checkRfidByMode(rfidMode: Int) {
        when (rfidMode) {
            RfidMode.DEFAULT -> defaultRfidRb!!.isChecked = true
            RfidMode.CUSTOM -> customRfidRb!!.isChecked = true
        }
    }

    object RfidMode {
        const val DEFAULT = 0
        const val CUSTOM = 1
    }

    private object CameraMode {
        const val SDK_SHOW_SCANNER = 0
        const val SHOW_CAMERA_ACTIVITY = 1
        const val SHOW_CUSTOM_CAMERA_ACTIVITY = 2
        const val SHOW_CUSTOM_CAMERA_ACTIVITY2 = 3
    }
}